webrtc.html

```diff
@@ -1251,30 +1251,99 @@
                 <p>If the <code>RTCPeerConnection</code> is configured to
                 generate Identity assertions by calling
                 <code>setIdentityProvider</code>, then the session description
-                SHALL contain an appropriate assertion. If the identity
-                provider is unable to produce an identity assertion, the call
-                to <code>createOffer</code> MUST be rejected with a
-                <code>DOMException</code> that has a name of
-                <code>NotReadableError</code>.</p>
-                <p>If this <code>RTCPeerConnection</code> object is closed
-                before the SDP generation process completes, the user agent
-                MUST suppress the result and not resolve or reject the returned
-                promise.</p>
-                <p>If the SDP generation process completed successfully, the
-                user agent MUST resolve the returned promise with a newly
-                created <code><a>RTCSessionDescription</a></code> object,
-                representing the generated offer.</p>
+                SHALL contain an appropriate assertion.</p>
                 <p class="fingerprint">The SDP generation process exposes a
                 subset of the media capabilities of the underlying system,
                 which provides generally persistent cross-origin information on
                 the device. It thus increases the fingerprinting surface of the
                 application. In privacy-sensitive contexts, browsers can
                 consider mitigations such as generating SDP matching only a
                 common subset of the capabilities.</p>
-                <p>If the SDP generation process failed for any other reason,
-                the user agent MUST reject the returned promise with an
-                <code>DOMException</code> object of type
-                <code>OperationError</code> as its argument.</p>
+                <p>When the method is called, the User Agent MUST run the
+                following steps:</p>
+                <ol>
+                  <li>
+                    <p>Let <var>connection</var> be the
+                    <code><a>RTCPeerConnection</a></code> object on which the
+                    method was invoked.</p>
+                  </li>
+                  <li>
+                    <p>Return the result of <a href=
+                    "#enqueue-an-operation">enqueuing</a> the following
+                    steps:</p>
+                    <ol>
+                      <li>
+                        <p>If <var>connection</var>'s [[<a>isClosed</a>]] slot
+                        is <code>true</code>, return a promise rejected with an
+                        <code>InvalidStateError</code>.</p>
```

Comment:
This text is in line with the queued steps of addIceCandidate, but I think what you're saying makes sense. So we check for closed state right away, and then make sure that nothing is taken of the queue if the isClosed slot is true.